Artifacts/windows-iis/artifact.ps1

```diff
@@ -0,0 +1,34 @@
+$artifactLogFolder = Join-path -path $env:ProgramData -childPath 'DTLArt_IIS'
+New-Item -ItemType Directory -Force -Path $artifactLogFolder
+
+$winVerString = (Get-WmiObject Win32_OperatingSystem).Caption
+Write-Output "Detected Windows Version: $winVerString"
+$scriptFolder = Split-Path $Script:MyInvocation.MyCommand.Path
+
+if ($winVerString -like 'Microsoft Windows Server 2008 R2*' -or $winVerString -like 'Microsoft Windows Server 2012*'){
+    Write-Output 'Installing IIS on Windows Server 2008 or 2012'
+	$localLog = Join-Path $artifactLogFolder 'IISArtifact.log'
+    $pkgMgrExe = join-path $env:windir system32\pkgmgr.exe
+	$packages = 'IIS-WebServerRole;IIS-WebServer;IIS-IIS6ManagementCompatibility;IIS-Metabase;IIS-WMICompatibility;IIS-LegacyScripts;IIS-LegacySnapIn'
+    $pkgMgrArgs = "/l:$localLog /iu:$packages"
+    $pkgMgrProcess = Start-Process -FilePath $pkgMgrExe -ArgumentList $pkgMgrArgs -Wait -PassThru
+
+    if($pkgMgrProcess.ExitCode -eq 0){
```

Comment:
Also check for reboot request scenarios (i.e. 3010).

Reply:
Added a message to the output if we encounter 3010